### PR TITLE
virt-controller, vm: Set domain firmware UUID based on namespace & name

### PR DESCRIPTION
**Pull request by EdDev**
### What this PR does

The UUID generated needs to depend on the uniqueness of the VM name and its namespace (same name may appear in two different namespaces).

fix: https://github.com/kubevirt/kubevirt/issues/11166

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/vm.go [hunk 1]
```diff
@@ -1824,7 +1824,7 @@ func setupStableFirmwareUUID(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachi
 		return
 	}
 
-	vmi.Spec.Domain.Firmware.UUID = types.UID(uuid.NewSHA1(firmwareUUIDns, []byte(vmi.ObjectMeta.Name)).String())
+	vmi.Spec.Domain.Firmware.UUID = types.UID(uuid.NewSHA1(firmwareUUIDns, []byte(vmi.Namespace+vmi.Name)).String())
```
Comment (vasiliy-ul) [inline pkg/virt-controller/watch/vm.go:1827]: :+1: 

Would it make sence to add a unit test? Perhaps extending this one:

https://github.com/kubevirt/kubevirt/blob/f2fcb8c57ee71b0440746f2bda4f528d279fcf83/pkg/virt-controller/watch/vm_test.go#L2053

Comment (EdDev) [reply pkg/virt-controller/watch/vm.go:1827]: Done

Comment (vasiliy-ul) [reply pkg/virt-controller/watch/vm.go:1827]: Thanks!

**Review by vasiliy-ul — APPROVED** [review]
/approve

**Pull request merged**